src/handle_check_address.c

```diff
@@ -18,76 +19,45 @@ void handle_check_address(check_address_parameters_t* params, chain_config_t* ch
     const uint8_t* bip32_path_ptr = params->address_parameters;
     uint8_t bip32PathLength = *(bip32_path_ptr++);
     cx_sha3_t local_sha3;
-
-    // Common memory is used for locals that are not used concurrently
-    union group1 {
-        uint32_t bip32Path[MAX_BIP32_PATH];
-        cx_ecfp_private_key_t privateKey;
-        char address[51];
-    } locals_union1;
-    union group2 {
-        uint8_t privateKeyData[64];
-        cx_ecfp_public_key_t publicKey;
-    } locals_union2;
+    uint32_t bip32Path[MAX_BIP32_PATH];
+    char address[51];
+    cx_ecfp_256_private_key_t priv;
+    cx_ecfp_256_public_key_t pub;
```

Comment:
IIRC we had to do the union optimization or we had stack issues :/

Reply:
After using `crypto_helpers.h`, we don't need exactly the same parameters.
Before the change, we had 2 `union` groups, but now, due with the new function calls, only `address[51]` can be mixed with `eitherbip32Path` or `priv` parameters. All other combinations are impossible because the parameters are used simultaeously.
Considering the sizes, we could use:
```
    union group {
        cx_ecfp_256_private_key_t priv;
        char address[51];
    } locals_union;
```

Do you think it is absolutely necessary?

Also, I have modified `cx_ecfp_private_key_t`/`cx_ecfp_public_key_t` by `cx_ecfp_256_private_key_t`/`cx_ecfp_256_public_key_t`, because it was already done like this in previous porting tasks. Which do you think is better (they are the same today)? Personnally, I prefer the version `xx_256_xx`, which is more explicit...

Reply:
> Do you think it is absolutely necessary?

Mmh this should be ok, address handling is not the most ram intensive task I think

cx_ecfp_256_private_key_t/cx_ecfp_256_public_key_t is fine 👍